### ISSUE TITLE
Option-left/right do not reposition the cursor correctly

### ISSUE THREAD

**Issue by ianthehenry** [zed-industries/zed#5331]
**Describe the bug**
Alt-left/right behave differently than native macOS text editing controls.

**To reproduce**
Text like this (pipe is cursor):

    hello.|

Press alt-left. See that the cursor is here:

    hello|.

**Expected behavior**

Cursor should be here:

    |hello.

**Environment:**

```
Zed 0.42.0 – /Applications/Zed.app
macOS 12.3.1
architecture x86_64
```

**Comment by notpeter** [comment]
I don't think we want to match macOS by default, but perhaps we can offer that as an option to support multiple word boundary movement types.
